src/addrman.h

```diff
@@ -317,7 +318,7 @@ class CAddrMan
 
         uint8_t compat;
         s >> compat;
-        const uint8_t lowest_compatible = compat - INCOMPATIBILITY_BASE;
+        const uint8_t lowest_compatible = std::max(compat - INCOMPATIBILITY_BASE, 0);
```

Comment:
Can you explain why treating corrupt files as file format version 0 is the right fix?

Reply:
Good point, it should raise.
```diff
-        if (lowest_compatible > FILE_FORMAT) {
+        if (lowest_compatible > FILE_FORMAT || compat < INCOMPATIBILITY_BASE) {
```